ci_framework/roles/ci_nmstate/library/ci_nmstate_apply_state.py

```diff
@@ -0,0 +1,152 @@
+  ---
+  module: ci_nmstate_apply_state
+  short_description: Apply a given nmstate state to the target host.
+  description:
+  - Apply a given nmstate state.
+  options:
+    state:
+      description:
+        - The nmstate state to apply.
+        - The content set here must follow the nmstate API U(https://nmstate.io/devel/api.html)
+      required: True
+      type: raw
+  requirements:
+    - libnmstate
+  author:
+    - Pablo Rodriguez (@pablintino)
+"""
+
+EXAMPLES = r"""
+  # Apply some network config to eth0 using nmstate
+  - name: Generate env var fact
+    register: get_makefiles_env_out
+    get_makefiles_env:
+      state:
+        dns-resolver:
+          running:
+            server:
+              - 10.111.222.1
+        routes:
+          config:
+          - destination: 0.0.0.0/0
+            next-hop-address: 10.111.222.1
+            next-hop-interface: eth0
+        interfaces:
+          - description: default
+            ipv4:
+              address:
+              -   ip: 10.111.222.33
+                  prefix-length: 24
+              dhcp: false
+              enabled: true
+            ipv6:
+              enabled: false
+            mtu: 1500
+            name: eth0
+            state: up
+            type: ethernet
+"""
+
+RETURN = r"""
+  state:
+    description: The nmstate state after applying the given state.
+    type: raw
+    returned: success
+    sample:
+      dns-resolver:
+        running:
+            server:
+            - 10.111.222.1
+      routes:
+        config:
+        - destination: 0.0.0.0/0
+          next-hop-address: 10.111.222.1
+          next-hop-interface: eth0
+      hostname:
+        config: test.node.lab.local
+        running: test.node.lab.local
+      interfaces:
+        - description: default
+          identifier: name
+          ipv4:
+            address:
+            -   ip: 10.111.222.33
+                prefix-length: 24
+            auto-dns: true
+            auto-gateway: true
+            auto-route-table-id: 0
+            auto-routes: true
+            dhcp: false
+            enabled: true
+          ipv6:
+            dhcp: false
+            enabled: false
+          mac-address: FA:16:3E:0A:20:87
+          mtu: 1500
+          name: eth0
+          profile-name: System eth0
+          state: up
+          type: ethernet
+"""
+
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+try:
+    from libnmstate import netapplier
+    from libnmstate import netinfo
+    from libnmstate.error import NmstateError
+
+    HAS_LIBNMSTATE = True
+except ImportError:
+    HAS_LIBNMSTATE = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec={
+            "state": {"required": True, "type": "raw"},
+        },
+        supports_check_mode=False,
+    )
+
+    if not HAS_LIBNMSTATE:
+        module.fail_json(
+            msg='Could not import "libnmstate" library. libnmstate is required on PYTHONPATH to run this module',
```

Comment:
Thank you for adding this for allowing users to install python3-libnmstate package. nmstate_unmanaged_provision_node.yml task file will take care of installing it.